skillmap/src/lib/browserUtils.ts

```diff
@@ -73,10 +57,53 @@ export async function getMarkdownAsync(source: MarkdownSource, url: string): Pro
 
     return {
         text: markdown,
-        url: toFetch
+        identifier: toFetch
     };
 }
 
+async function fetchSkillMapFromGithub(path: string): Promise<MarkdownFetchResult | undefined> {
+    const ghid = pxt.github.parseRepoId(path)
+    const config = await pxt.packagesConfigAsync();
+    const status = pxt.github.repoStatus(ghid, config);
+
+    let reportId: string | undefined;
+    switch (status) {
+        case pxt.github.GitRepoStatus.Banned:
+            throw lf("This GitHub repository has been banned.");
+        case pxt.github.GitRepoStatus.Approved:
+            reportId = undefined;
+            break;
+        default:
+            reportId = "https://github.com/" + ghid.fullName;
+            break;
+    }
+
+    const tag = ghid.tag || await pxt.github.latestVersionAsync(ghid.fullName, config, true);
+
+    if (!tag) {
+        pxt.log(`skillmap github tag not found at ${ghid.fullName}`);
+        return undefined;
+    }
+    ghid.tag = tag;
+
+    const gh = await pxt.github.downloadPackageAsync(`${ghid.fullName}#${ghid.tag}`, config);
+
+    if (gh) {
+        let fileName = ghid.fileName ||  "skillmap";
+        fileName = fileName.replace(/^\/?blob\/main\//, "")
+        fileName = fileName.replace(/^\/?blob\/master\//, "")
+        fileName = fileName.replace(/\.md$/, "")
+
+        return {
+            text: pxt.tutorial.resolveLocalizedMarkdown(ghid, gh.files, fileName),
+            identifier: ghid.fullName + "#" + fileName,
```

Comment:
maybe a comment describing the identifier format somewhere? (it's "user/repo#file" i'm assuming?)